integrationtests/src/test/java/io/kroxylicious/proxy/KroxyConfigBuilderTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class KroxyConfigBuilderTest {
```

Comment:
I'm not a fan of the tests asserting the entire YAML document as that is a very brittle test that will fail if we make any changes to the structure (such as omitting empty blocks)

I think I'd like to see some round trip assertions where the config is parsed back to the builder and we assert the values are as expected.

Reply:
:+1: I've refactored to use Jackson to deserialize back to JsonNode so we make less brittle assertions

Reply:
I've not implemented any deserialization test since we only need one way serialization from object -> yaml.